samza-core/src/main/java/org/apache/samza/zk/ZkCoordinationServiceFactory.java

```diff
@@ -19,32 +19,57 @@
 package org.apache.samza.zk;
 
 import com.google.common.base.Strings;
+import java.util.Map;
 import org.I0Itec.zkclient.ZkClient;
 import org.apache.samza.SamzaException;
 import org.apache.samza.config.Config;
+import org.apache.samza.config.MetricsConfig;
 import org.apache.samza.config.ZkConfig;
 import org.apache.samza.coordinator.CoordinationServiceFactory;
 import org.apache.samza.coordinator.CoordinationUtils;
+import org.apache.samza.metrics.MetricsRegistryMap;
+import org.apache.samza.metrics.MetricsReporter;
+import org.apache.samza.util.MetricsReporterLoader;
 import org.apache.zookeeper.client.ConnectStringParser;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 
 public class ZkCoordinationServiceFactory implements CoordinationServiceFactory {
-  private final static Logger LOG = LoggerFactory.getLogger(ZkCoordinationServiceFactory.class);
+  private static final Logger LOG = LoggerFactory.getLogger(ZkCoordinationServiceFactory.class);
+
+  private static final String SOURCE_PREFIX = "ZkJobCoordinationService";
 
   public CoordinationUtils getCoordinationService(String groupId, String participantId, Config config) {
     ZkConfig zkConfig = new ZkConfig(config);
 
     ZkClient zkClient =
         createZkClient(zkConfig.getZkConnect(), zkConfig.getZkSessionTimeoutMs(), zkConfig.getZkConnectionTimeoutMs());
 
-    ZkUtils zkUtils = new ZkUtils(new ZkKeyBuilder(groupId), zkClient, zkConfig.getZkConnectionTimeoutMs());
+    ZkJobCoordinatorMetrics zkJobCoordinatorMetrics = getJobCoordinatorMetrics(config, participantId);
+
+    ZkUtils zkUtils = new ZkUtils(new ZkKeyBuilder(groupId), zkClient, zkConfig.getZkConnectionTimeoutMs(), zkJobCoordinatorMetrics);
```

Comment:
>Q1: 

Agree with you. We could tie up the reporters lifecycle with ZkCoordinatorService. We can start reporters in ZkCoordinatorService constructor and stop reporters in reset(tieing reporters in ZkCoordinationService is synonymous to ZkJobCoordinator approach of reporter lifecycle management).

>Q2: 

I was with wrong notion that this is synonymous to processorID. There doesn't seem to be obvious way to address the random participant problem(Utilizing processorId for this doesn't look logical at LocalApplicationRunner level. Currently all stream application have same app id, app name. If acceptable, we can report metrics with (appid,appname) & use hostname to solve this problem partially and create a JIRA for proper solution later).